packages/core/src/errors.ts

```diff
@@ -12,9 +12,22 @@ export enum ProviderFailure {
 
 const formatMessage = (reason: string, detail?: string) => reason + (detail ? ` (${detail})` : '');
 
+const tryToUnhideInnerError = (error: Error, innerError: unknown) => {
```

Comment:
How about creating a new error base class that would do this? Then we wouldn't need to call this for every single error  type.

Reply:
Looks good @iccicci . Please search for other `extends CustomError` within sdk and update those as well